packages/mail-client/src/mail-client.spec.ts

```diff
@@ -0,0 +1,47 @@
+import nodemailer from 'nodemailer';
+import { sendMail } from './mail-client';
+import { MailOptions } from './mail-client-types';
+
+describe('mail client', () => {
+  class MockMailer {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    sendMail() {}
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    close() {}
+  }
+  it('should create transport, send mails and close the transport', async () => {
+    const spyCreateTransport = jest
+      .spyOn(nodemailer, 'createTransport')
+      .mockReturnValue(new MockMailer() as any);
+    const spySendMail = jest.spyOn(MockMailer.prototype, 'sendMail');
+    const spyClose = jest.spyOn(MockMailer.prototype, 'close');
+    const destination: any = {
+      originalProperties: {
+        'mail.password': 'password',
+        'mail.user': 'user',
+        'mail.smtp.host': 'smtp.gmail.com',
+        'mail.smtp.port': '587'
+      },
+      name: 'my-destination',
+      type: 'MAIL',
+      authentication: 'BasicAuthentication',
+      proxyType: 'Internet'
+    };
+    const mailOptions1: MailOptions = {
+      from: 'from1@example.com',
+      to: 'to1@example.com'
+    };
+    const mailOptions2: MailOptions = {
+      from: 'from2@example.com',
+      to: 'to2@example.com'
+    };
+    await expect(
+      sendMail(destination, mailOptions1, mailOptions2)
```

Comment:
[q] can I also put a `[mail1,mail2]` in the sendMail or will this fail? I think we had a similar issue in the past. We have to keep in mind the big JS user base how do not get type errors so the best thing in my opinion would to overload either `sendMail(destination,mail)` and `sendMail(destiation,[mail1,mail2])` The case with the variadic argument I find not really useful to be honest.